api/handlers/service_instance_test.go

```diff
@@ -73,6 +73,294 @@ var _ = Describe("ServiceInstance", func() {
+				})
+
+				When("the service instance is managed", func() {
+					BeforeEach(func() {
+						serviceInstanceRepo.GetServiceInstanceReturns(repositories.ServiceInstanceRecord{
+							GUID: "service-instance-guid", Type: korifiv1alpha1.ManagedType, PlanGUID: "service-plan-guid",
+						}, nil)
+					})
+
+					It("includes broker fields in the response", func() {
+						Expect(rr).Should(HaveHTTPStatus(http.StatusOK))
+						Expect(rr).To(HaveHTTPBody(SatisfyAll(
+							MatchJSONPath("$.included.service_brokers[0].guid", "service-broker-guid"),
+							MatchJSONPath("$.included.service_brokers[0].name", "service-broker-name"),
+						)))
+					})
+				})
+			})
+		})
+	})
+
+	Describe("GET /v3/service_instances/:guid/credentials", func() {
+		BeforeEach(func() {
+			serviceInstanceRepo.GetServiceInstanceReturns(repositories.ServiceInstanceRecord{
+				GUID: "service-instance-guid",
+				Type: korifiv1alpha1.UserProvidedType,
+			}, nil)
+
+			serviceInstanceRepo.GetServiceInstanceCredentialsReturns(map[string]any{
+				"foo": "bar",
+			}, nil)
+
+			reqPath += "/service-instance-guid/credentials"
+		})
+
+		It("gets the service instance credentials", func() {
+			Expect(serviceInstanceRepo.GetServiceInstanceCallCount()).To(Equal(1))
+			_, actualAuthInfo, actualServiceInstanceGUID := serviceInstanceRepo.GetServiceInstanceArgsForCall(0)
+			Expect(actualAuthInfo).To(Equal(authInfo))
+			Expect(actualServiceInstanceGUID).To(Equal("service-instance-guid"))
+
+			Expect(serviceInstanceRepo.GetServiceInstanceCredentialsCallCount()).To(Equal(1))
+			_, actualAuthInfo, actualInstanceGUID := serviceInstanceRepo.GetServiceInstanceCredentialsArgsForCall(0)
+			Expect(actualAuthInfo).To(Equal(authInfo))
+			Expect(actualInstanceGUID).To(Equal("service-instance-guid"))
+
+			Expect(rr).Should(HaveHTTPStatus(http.StatusOK))
+			Expect(rr).To(HaveHTTPHeaderWithValue("Content-Type", "application/json"))
+			Expect(rr).To(HaveHTTPBody(SatisfyAll(
+				MatchJSONPath("$.foo", "bar"),
+			)))
+		})
+
+		When("the service instance is not user-provided", func() {
+			BeforeEach(func() {
+				serviceInstanceRepo.GetServiceInstanceReturns(repositories.ServiceInstanceRecord{
+					GUID: "service-instance-guid",
+					Type: korifiv1alpha1.ManagedType,
+				}, nil)
+			})
+
+			It("returns a 404 Not Found error", func() {
+				expectNotFoundError("Service Instance")
+			})
+		})
+
+		When("the service instance does not have credentials", func() {
+			BeforeEach(func() {
+				serviceInstanceRepo.GetServiceInstanceReturns(repositories.ServiceInstanceRecord{
+					GUID: "service-instance-guid",
+					Type: korifiv1alpha1.UserProvidedType,
+				}, nil)
+
+				serviceInstanceRepo.GetServiceInstanceCredentialsReturns(map[string]any{}, apierrors.NewNotFoundError(nil, repositories.ServiceInstanceResourceType))
+			})
+
+			It("returns 404 Not Found", func() {
+				_, actualAuthInfo, actualInstanceGUID := serviceInstanceRepo.GetServiceInstanceCredentialsArgsForCall(0)
```

Comment:
From my point of view, checking `GetServiceInstanceCredentials` invocation and its parameter [duplicates the checks](https://github.com/cloudfoundry/korifi/pull/3610/files#diff-7e9693740ebe688850df0973f1a6dfd8480f56cd87d4e58e4fadb1fe09929a73R285-R287) in the default case. I think expecting a not found error is enough for this test.